InformationScripting/scripts/color.py

```diff
@@ -8,6 +8,6 @@
 for ts in inputs:
     tuples = ts.take("ast")
     for tuple in tuples:
-        tuple.add(NamedProperty("color", color))
-        ts.add(tuple)
+        t = Tuple([NamedProperty("color", color), NamedProperty("ast", tuple.ast)])
```

Comment:
Tuples are just immutable lists in python. They can contain any type. So we can not guarantee it has size 2 and neither that the first value is a string. If we explicitly have to type NamedProperty this guarantee is given.

On the other hand we can just throw an exception if the tuple has the wrong form.